### DISCUSSION TITLE
lang option not working in defaults file

### DISCUSSION THREAD

**Discussion by jskvbinmv** [Zettlr/Zettlr#3558] [Q&A]
Hello. 

First of all thanks for the great tool!

I have a problem when setting the lang variable in the defaults file. It works perfectly using a YAML Frontmatter  but when I put the same line in the defaults file (via Assets manager) I get the following error:

`Aeson exception:
Error in $: Unknown option "lang"`

My defaults file is the one that ships with Zettlr except for the one line:

`lang: de-DE`

When I add the YAML frontmatter

```
---
lang: de-DE
...
```
the export works without any error. 

What am I doing wrong?

**Comment by nathanlesage** [comment]
The lang option is simply not supported in the defaults file, only in the front matter. The language is a part of the document's metadata, whereas the defaults files rather set more broad instructions.

**Reply by jskvbinmv** [reply]
Thanks. But in this case it is odd that setting the lang option in the defaults file is the option mentioned as a common use case of defaults files in the documentation.

https://docs.zettlr.com/en/core/defaults-files/

> One common use-case for defining variables inside a defaults file which you could also define on the level of YAML frontmatters would be the lang-property.

**Reply by nathanlesage** [reply]
Ah, I have to admit that I lied. Of course you can add frontmatter variables also to defaults files. HOWEVER, you will have to wrap them into a `metadata` property. See the screenshot:

![image](https://user-images.githubusercontent.com/17251683/174831648-663d78da-6559-4882-aa67-c45497a64017.png)

**Reply by jskvbinmv** [reply]
Thank you! That worked.

Is there a list, which variables are considered metadata?

**Reply by nathanlesage** [reply]
Not yet, but I plan on writing a comprehensive guide on that mess probably this week
